lib/flapjack/gateways/hipchat.rb

```diff
@@ -0,0 +1,123 @@
+        @config = opts[:config]
+        @logger = opts[:logger]
+        @redis_config = opts[:redis_config] || {}
+        @redis = Flapjack::RedisPool.new(:config => @redis_config, :size => 1, :logger => @logger)
+
+        @logger.info("starting")
+        @logger.debug("new hipchat gateway pikelet with the following options: #{@config.inspect}")
+
+        @sent = 0
+      end
+
+      def stop
+        @logger.info("stopping")
+        @should_quit = true
+
+        redis_uri = @redis_config[:path] ||
+          "redis://#{@redis_config[:host] || '127.0.0.1'}:#{@redis_config[:port] || '6379'}/#{@redis_config[:db] || '0'}"
+        shutdown_redis = EM::Hiredis.connect(redis_uri)
+        shutdown_redis.rpush(@config['queue'], Flapjack.dump_json('notification_type' => 'shutdown'))
+      end
+
+      def start
+        queue = @config['queue']
+
+        until @should_quit
+          begin
+            @logger.debug("hipchat gateway is going into blpop mode on #{queue}")
+            alert = Flapjack::Data::Alert.next(queue, :redis => @redis, :logger => @logger)
+            deliver(alert) unless alert.nil?
+          rescue => e
+            @logger.error "Error generating or dispatching Hipchat message: #{e.class}: #{e.message}\n" +
+              e.backtrace.join("\n")
+          end
+        end
+      end
+
+      def deliver(alert)
+        username     = @config["username"]
+      	api_token    = @config["api_token"]
+      	room         = @config["room"]
+
+        notification_id = alert.notification_id
+        message_type    = alert.rollup ? 'rollup' : 'alert'
+
+        hipchat_template_erb, hipchat_template =
+          load_template(@config['templates'], message_type, 'text',
+                        File.join(File.dirname(__FILE__), 'hipchat'))
+
+        @alert  = alert
+        bnd     = binding
+
+        begin
+          message = hipchat_template_erb.result(bnd).chomp
+        rescue => e
+          @logger.error "Error while executing the ERB for a hipchat message: " +
+            "ERB being executed: #{hipchat_template}"
+          raise
+        end
+
+        errors = []
+
+        [  
+         [username, "Hipchat username is missing"],  
+         [api_token, "Hipchat api_token is missing"],
+         [room, "Hipchat room is missing"]
+        ].each do |val_err|
+
+          next unless val_err.first.nil? || (val_err.first.respond_to?(:empty?) && val_err.first.empty?)
+          errors << val_err.last
+        end
+
+        unless errors.empty?
+          errors.each {|err| @logger.error err }
+          return
+        end
+        
+        payload = {
+          :message_format    => 'html',
+          :color             => @alert.state == 'ok' ? 'green' : 'red'
+        }
+        
+        reported_payload = payload.merge(:message => message)
+        
+        @logger.debug "payload: #{reported_payload}"
+        
+        client = HipChat::Client.new(api_token, :api_version => 'v2')
```

Comment:
Could the client be initialized on pikelet setup rather than every time a message is delivered?